pynvim/util.py

```diff
@@ -40,3 +41,22 @@ def get_client_info(kind, type_, method_spec):
 
 
 VERSION = Version(major=0, minor=3, patch=2, prerelease='')
+
+
+def transform_keyerror(f):
+    """Decorator to transform "Key not found" NvimErrors into KeyErrors."""
+    @functools.wraps(f)
+    def wrapper(*args, **kwargs):
+        # XXX: # not on top, because of circular imports.
+        from .api.nvim import NvimError, NvimKeyError
```

Comment:
Why not have this in the same module as `RemoteMap` where it is used, it should avoid the import issue? Also not sure this is made simpler/clearer by using a decorator, we could just write out `try/expect` in both `__getitem__` and `__delitem__` (and call a value-changing helper function).

Reply:
the import issue is for `NvimError`.
But it could be moved there maybe.

> and call a value-changing helper function

What do you mean?

Yes, using a decorator is not really necessary.

Reply:
I mean we could still factor out the check of the error value to a helper function (i e check if the string matches, and return a different error object).

Reply:
ok, done.